### PR TITLE
Fixes #11255 - Fixed checks for explicit input in host cloning

### PR DESCRIPTION
**Pull request by ShimShtein**
Now clone action will try to inherit values from the hostgroup, unless the user explicitly unchecked the "inherit" button. Custom errors added to each field that will be overridden by this action.

### CODE REVIEW

**Comment by domcleal** [comment]
I don't think the behaviour to automatically inherit is correct in the clone case, since you specifically want to create a host that's as close as possible in behaviour to the original.  Inheriting attributes from the host group should require an additional action/change.

Is it possible to change the form to _not_ inherit by default, and to lose the error messages/tips?

**Comment by ShimShtein** [comment]
@domcleal That's true, but I don't have a good way of distinguishing between a newly created host and a cloned one. I can look at the request action of course, but it doesn't sound right, as it creates another list of actions that could be forgotten by a programmer.
Any ideas for a better `should_inherit?` logic?

**Comment by domcleal** [comment]
The `@clone_host` instance var's set from the clone action, if that helps?

**Comment by ShimShtein** [comment]
@domcleal, it would create just another edge case. I'm looking for a more generic rules... In the worst case I will add a test for `@clone_host`, but I don't really like it.

**Comment by ShimShtein** [comment]
@domcleal fixed by inspecting the action. Can't say it's the best solution, but I can't think of a better solution.

File: app/helpers/hosts_and_hostgroups_helper.rb [hunk 1]
```diff
@@ -109,4 +109,10 @@ def interesting_klasses(obj)
 
     classes.where(:id => klasses)
   end
+
+  def explicit_value?(field)
+    return true if params[:action] == 'clone'
+    return unless params[:host]
```
Comment (domcleal) [inline app/helpers/hosts_and_hostgroups_helper.rb]: Nitpick, return false instead of nil for consistency with the other possible bool return values.  (Ditto in the other method)

**Comment by domcleal** [comment]
That's okay, I think it's passable.

Would you mind adding a basic functional test?  I think one similar to HostsControllerTest#test_clone that simply ensures that `get :clone` renders successfully with an FG host with `:with_hostgroup` would reproduce the bug.

**Comment by ShimShtein** [comment]
@domcleal  Fixed syntax and test added.

**Comment by domcleal** [comment]
Thanks @ShimShtein , merged as 5081efd.
